### ISSUE TITLE
Web requests should not be automatically retried after 120 seconds when they are still pending.

### ISSUE THREAD

**Issue by CedricMalfreytBL** [cypress-io/cypress#9219]
### Current behavior

Web requests are automatically retried after 120 seconds even if they are still pending.

### Desired behavior

Don't retry web requests that are still pending.

### Test code to reproduce

Refer to https://github.com/cypress-io/cypress/issues/2055

Issue 2055 is still active and adding the line `this._server.setTimeout(0);` (see bellow) to `server.js` resolves the issue :

```js
  createServer (app, config, project, request, onWarning) {
    return new Promise((resolve, reject) => {
      const { port, fileServerFolder, socketIoRoute, baseUrl } = config

      this._server = http.createServer(app)
      this._server.setTimeout(0); // added line
      allowDestroy(this._server)

      const onError = (err) => {
        // if the server bombs before starting
        // and the err no is EADDRINUSE
        // then we know to display the custom err message
        if (err.code === 'EADDRINUSE') {
          return reject(this.portInUseErr(port))
        }
      }
```

Please, add that line.

### Versions

5.6.0

**Comment by jennifer-shehane** [comment]
Can you provide a reproducible example of a situation where the requests are being retried in Cypress at 120s and not in Chrome normally? There are some situations where Chrome will retry requests at 120s, so we'd need to see your specific example in order to determine if this is desired behavior (matching the browser behavior) or if it is behaving differently within Cypress.

**Comment by CedricMalfreytBL** [comment]
Hi Jennifer, of course.

This is a full example :
[issue9219.zip](https://github.com/cypress-io/cypress/files/5555724/issue9219.zip)

How to use :
- npm install
- npm run back
- npm run front

Open your browser and navigate to http://localhost:3321/

You can test : fill the desired timeout (in ms) in the input. You can try 5000 or 125000, there will be no problem : one request and one response.
![Chrome](https://user-images.githubusercontent.com/8135706/99438404-0c74d480-2914-11eb-93d4-3a86e9d27c97.png)


Then run "npm run test", wait for Cypress to open and run the test, which fill 125000 in the timeout.
At 120 seconds, a new request is sent, and at 125 seconds, the response for the first request is sent, but the test continues because the second resuest is still running.
![Cypress](https://user-images.githubusercontent.com/8135706/99438425-14cd0f80-2914-11eb-8406-88dd8e9225eb.png)


Now, modify server.js (i don't provide that, I guess you fave it) and add the line this._server.setTimeout(0) :

  createServer (app, config, project, request, onWarning) {
    return new Promise((resolve, reject) => {
      const { port, fileServerFolder, socketIoRoute, baseUrl } = config

      this._server = http.createServer(app)
      this._server.setTimeout(0); // added line
      allowDestroy(this._server)

      const onError = (err) => {
        // if the server bombs before starting
        // and the err no is EADDRINUSE
        // then we know to display the custom err message
        if (err.code === 'EADDRINUSE') {
          return reject(this.portInUseErr(port))
        }
      }


Run the test again, the behavior will be like in a browser.

Don't forget to modify for https.

**Comment by CedricMalfreytBL** [comment]
I've downloaded the standalone version here : https://download.cypress.io/desktop and applied the patch.

This is a screenshot of the result :
![Cypress_patched](https://user-images.githubusercontent.com/8135706/99444276-48139c80-291c-11eb-92c9-03e42679eb77.png)

No more issue.

**Comment by jennifer-shehane** [comment]
Running the code outside of Cypress actually did re-issue the request at 120s, with it returning the response twice. 🤔 I added timestamps to the logs. 

Maybe there's some difference in our environment that's affecting it?

### Outside Cypress

<img width="1098" alt="Screen Shot 2020-11-18 at 2 13 14 PM" src="https://user-images.githubusercontent.com/1271364/99500142-4ee7f100-29a8-11eb-8cdf-70302d25d203.png">

### In Cypress

<img width="1014" alt="Screen Shot 2020-11-18 at 2 24 09 PM" src="https://user-images.githubusercontent.com/1271364/99501177-cbc79a80-29a9-11eb-829f-7b922e989934.png">

**Comment by CedricMalfreytBL** [comment]
Hi Jennifer,

I work for an software enterprise and we already issue this behavior at some customers. It was always due to a proxy.

Be sure to have no proxy.

I'm on Windows 10 2004. Are you also testing on Windows 10 ?
My node version : 14.15.0
My npm version : 6.14.8 (shouldn't affect the test)

**Comment by jennifer-shehane** [comment]
Yeah, there is likely something that is causing the request to error when passed through a proxy - and the default behavior of Cypress is to retry at 120s when the request errors. 

I'd have to review with the engineers why they didn't implement the workaround of setting the timeout. I think there may have been a reason that they avoided adding this line.

What is the actual error you're encountering when testing? You have a request that definitely takes longer than 2 minutes? How critical is this situation in your testing?

**Comment by CedricMalfreytBL** [comment]
Hi Jennifer, thank you for investigating.

Having a request that takes longer that 2 minutes is definitly a bad behavior, I agree with that. The problem is that the tests on Cypress fails while there is no problem in the real life.

We should be able to rely on Cypress and have to be sure that its proxy doesn't produce false issues.

In addition, the actual behavior doesn't respect the requestTimeout and responseTimeout we can specify on cy.wait and this is in my opinion an issue as well.

To my mind the patch should be applied in Cypress as it solves an issue and I'm quite sure it won't produce other.

**Comment by CedricMalfreytBL** [comment]
Hi Jennifer,

Is there any update on this issue ?

**Comment by CedricMalfreytBL** [comment]
Hi Jennifer,

After updating to version 6.0.1 of Cypress, the issue still exists.

**Comment by CedricMalfreytBL** [comment]
Hi Jennifer,

After updating to version 6.1.0 of Cypress, the issue still exists.

**Comment by jennifer-shehane** [comment]
This issue is still in the 'proposal' stage, which means no work has been done on this issue as of today. We will comment in the issue when there are any updates, but right now this work is not a priority.

**Comment by pmammadli** [comment]
Having the same issue

**Comment by amac714** [comment]
any updates on this? Having the same issue

**Comment by CedricMalfreytBL** [comment]
I think @jennifer-shehane  should remove the "stage: needs information" tag in order for this issue to be solved by someone.
I think all informations needed are in this thread, right ?

**Comment by GreyMerchant** [comment]
Hi there, 

I understand this issue is closed for the moment but hasn't been addressed. Can someone provide me with a workaround for the moment please?

**Comment by CedricMalfreytBL** [comment]
Hi @GreyMerchant ,

Look at https://github.com/cypress-io/cypress/issues/9219#issuecomment-729155195 for solving the issue, so simple !

But it seems that Cypress team doesn't want to solve it, so we decided to stop using Cypress.

**Comment by ghost** [comment]
Hi @jennifer-shehane, still no updates on this ?

**Comment by nagash77** [comment]
@CedricMalfreytBL your last comment indicated you stopped using Cypress.  Are you still using Cypress?  If so are you still seeing this issue on the latest version of Cypress?

**Comment by ghost** [comment]
Hi @nagash77, the issue no more exists with the latest version (12.12.0).

**Comment by nagash77** [comment]
@CedricMalfreytBL , thank you for confirming that this is no longer an issue in the latest version of Cypress.  I will go ahead and close this issue at this point.
